rhel.Dockerfile

```diff
@@ -9,26 +9,28 @@
 #   Red Hat, Inc. - initial API and implementation
 #
 
-# UPSTREAM: use devtools/go-toolset-rhel7 image so we're not required to authenticate with registry.redhat.io
-# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/rhel8/go-toolset
-FROM registry.access.redhat.com/devtools/go-toolset-rhel7:1.11.13-11.1571302666 as builder
-ENV PATH=/opt/rh/go-toolset-1.11/root/usr/bin:$PATH
-
-# DOWNSTREAM: use rhel8/go-toolset; no path modification needed
-# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/rhel8/go-toolset
-# FROM registry.redhat.io/rhel8/go-toolset:1.12.8-18 as builder
-
-ENV GOPATH=/go/
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/devtools/go-toolset-rhel7
+FROM registry.access.redhat.com/devtools/go-toolset-rhel7:1.12.12-4 as builder
+ENV PATH=/opt/rh/go-toolset-1.12/root/usr/bin:$PATH \
+    GOPATH=/go/
 USER root
 WORKDIR /go/src/github.com/eclipse/che-jwtproxy/
-COPY . .
-RUN CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-w -s' -a -installsuffix cgo -o jwtproxy cmd/jwtproxy/main.go
+COPY . /go/src/github.com/eclipse/che-jwtproxy/
+RUN adduser appuser && \
+    CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-w -s' -a -installsuffix cgo -o jwtproxy cmd/jwtproxy/main.go
 
 # https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
-FROM registry.access.redhat.com/ubi8-minimal:8.1-328
+FROM registry.access.redhat.com/ubi8-minimal:8.1-398
 
+USER appuser
 ENV XDG_CONFIG_HOME=/che-jwtproxy-config/
 VOLUME /config
```

Comment:
Seems like inconsistency:

Dockerfile
```
VOLUME /che-jwtproxy-config
...
ENTRYPOINT ["jwtproxy"]
CMD ["-config", "/che-jwtproxy-config/config.yaml"]
```
Dockerfile.rhel
```
VOLUME /config
ENTRYPOINT ["jwtproxy"]
CMD ["-config", "/che-jwtproxy-config/config.yaml"]
```

I assume that Volume should not be here, but I'm not sure.
Then it even makes sense to miss CMD and require it from clients, like it will be more straight forward if Che Server sets it, and only it exactly know where volume mounts are bound.

Be careful while applying my suggestions since I don't know che-jwtproxy very well and my understanding may be incorrect.

Reply:
The difference in volumes is caused by my oversight in #19 
And I agree with you that we don't actually need the `VOLUME` and `CMD` here. I have changes locally that I want to introduce with the fix to the single user that simplify the image in the exact manner you suggest.

I just didn't want to introduce them just yet because they require changes on the server-side as well (namely configuring the pod to also declare the `XDG_CONFIG_HOME` env var).

As a sneak peak, this is what I have locally ATM:

```diff
diff --git a/Dockerfile b/Dockerfile
index f967c1a..f118ad3 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -17,8 +17,10 @@ RUN CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-w -s' -a -installsuffix cgo
 
 
 FROM alpine:3.7
-ENV XDG_CONFIG_HOME=/che-jwtproxy-config/
-VOLUME /che-jwtproxy-config
 COPY --from=builder /go/src/github.com/eclipse/che-jwtproxy/jwtproxy /usr/local/bin
 ENTRYPOINT ["jwtproxy"]
-CMD ["-config", "/che-jwtproxy-config/config.yaml"]
+# The JWT proxy needs 2 things:
+# * the location of the configuration file supplied as an argument:
+#   `-config <location/of/the/config.yaml>`
+# * The XDG_CONFIG_HOME environment variable pointing to a directory where to store auth keys
+# CMD ["-config", "/che-jwtproxy-config/config.yaml"]
diff --git a/rhel.Dockerfile b/rhel.Dockerfile
index 037bae0..e42ff61 100644
--- a/rhel.Dockerfile
+++ b/rhel.Dockerfile
@@ -27,10 +27,12 @@ RUN CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-w -s' -a -installsuffix cgo
 # https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
 FROM registry.access.redhat.com/ubi8-minimal:8.1-328
 
-ENV XDG_CONFIG_HOME=/config/
-VOLUME /config
 COPY --from=builder /go/src/github.com/eclipse/che-jwtproxy/jwtproxy /usr/local/bin
 ENTRYPOINT ["jwtproxy"]
-CMD ["-config", "/config/config.yaml"]
+# The JWT proxy needs 2 things:
+# * the location of the configuration file supplied as an argument:
+#   `-config <location/of/the/config.yaml>`
+# * The XDG_CONFIG_HOME environment variable pointing to a directory where to store auth keys
+# CMD ["-config", "/che-jwtproxy-config/config.yaml"]
 
 # append Brew metadata here
```